packages/client/src/impl/WriteApiImpl.ts

```diff
@@ -0,0 +1,116 @@
+import WriteApi from '../WriteApi'
+import {DEFAULT_WriteOptions, WriteOptions} from '../options'
+import {Transport} from '../transport'
+import {Headers} from '../results'
+import {Log} from '../util/logger'
+import {HttpError} from '../errors'
+
+export default class WriteApiImpl implements WriteApi {
+  private closed = false
+
+  constructor(private transport: Transport) {
+    this.doWrite = this.doWrite.bind(this)
+  }
+
+  _createWritePath(bucket: string, writeOptions: WriteOptions, org?: string) {
+    const query: string[] = [
+      `bucket=${encodeURIComponent(bucket)}`,
+      `precision=${writeOptions.precision}`,
+    ]
+    if (org) query.push(`org=${encodeURIComponent(org)}`)
+    const consistency = writeOptions?.consistency
+    if (consistency)
+      query.push(`consistency=${encodeURIComponent(consistency)}`)
+
+    const path = `/api/v2/write?${query.join('&')}`
+    return path
+  }
+
+  doWrite(
+    lines: string[],
+    bucket: string,
+    org?: string,
+    writeOptions?: Partial<WriteOptions>
+  ): Promise<void> {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self: WriteApiImpl = this
+    if (self.closed) {
+      return Promise.reject(new Error('writeApi: already closed!'))
+    }
+    if (lines.length <= 0 || (lines.length === 1 && lines[0] === ''))
+      return Promise.resolve()
+
+    let resolve: (value: void | PromiseLike<void>) => void
+    let reject: (reason?: any) => void
+    const promise = new Promise<void>((res, rej) => {
+      resolve = res
+      reject = rej
+    })
+
+    let responseStatusCode: number | undefined
+    const callbacks = {
+      responseStarted(_headers: Headers, statusCode?: number): void {
+        responseStatusCode = statusCode
+      },
+      error(error: Error): void {
+        // ignore informational message about the state of InfluxDB
+        // enterprise cluster, if present
+        if (
+          error instanceof HttpError &&
+          error.json &&
+          typeof error.json.error === 'string' &&
+          error.json.error.includes('hinted handoff queue not empty')
+        ) {
+          Log.warn('Write to InfluxDB returns: ' + error.json.error)
```

Comment:
Please use template string:

```suggestion
          Log.warn(`Write to InfluxDB returns: ${error.json.error}`)
```

Reply:
Forcing template string is now added into eslintrc

Reply:
Nice! 👍